File: my_cpp_pkg/src/add_two_ints_test_client.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/srv/add_two_ints.hpp>

using namespace std::chrono_literals;

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  auto pNode = std::make_shared<rclcpp::Node>("add_two_ints_test_client");
  auto pClient = pNode->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");

  while (!pClient->wait_for_service(1s))
  {
    RCLCPP_WARN(pNode->get_logger(), "Waiting for Add Two Ints Server...");
  }

  auto pRequest = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
  pRequest->a = 4;
  pRequest->b = 7;

  auto future_response = pClient->async_send_request(pRequest);
  rclcpp::spin_until_future_complete(pNode, future_response);

  auto pResponse = future_response.get();

  RCLCPP_INFO(pNode->get_logger(), "%ld + %ld = %ld", pRequest->a, pRequest->b, pResponse->sum);

  rclcpp::shutdown();
  return 0;
}

File: my_py_pkg/my_py_pkg/add_two_ints_test_client.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


def main(args=None):
    rclpy.init(args=args)

    node = Node("add_two_ints_test_client")

    client = node.create_client(srv_type=AddTwoInts, srv_name="add_two_ints")

    while not client.wait_for_service(timeout_sec=1):
        node.get_logger().warn("Waiting for Add Two Ints Server...")

    request = AddTwoInts.Request()
    request.a = 4
    request.b = 7

    future_response = client.call_async(request)
    rclpy.spin_until_future_complete(node, future_response)

    response = future_response.result()

    node.get_logger().info(
        str(request.a) + " + " + str(request.b) + " = " + str(response.sum)
    )

    rclpy.shutdown()


if __name__ == "__main__":
    main()
